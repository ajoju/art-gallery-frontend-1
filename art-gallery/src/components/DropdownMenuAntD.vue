<template>
    <div class="nav-site">
        <a-menu v-model:selectedKeys="current" mode="horizontal"  class="nav-menu">
            <a-menu-item key="home">
                <router-link to="/" class="nav-link">
                    <span>Home</span>
                </router-link>
            </a-menu-item>
            <a-sub-menu>
                <template #title>
                    <router-link to="/artworks" class="nav-link">
                        <span>Artworks</span>
                    </router-link>
                </template>
                <a-menu-item key="artwork-of-day" class="menu-item">
                    <router-link to="/artworkofday" class="nav-sub">
                        <span>Artwork of the Day</span>
                    </router-link>
                </a-menu-item>
                <a-menu-item key="artworks-list">
                    <router-link to="/artworks" class="nav-sub">
                        <span>List of Artworks</span>
                    </router-link>
                </a-menu-item>
            </a-sub-menu>
            <a-sub-menu>
                <template #title>
                    <router-link to="/culture" class="nav-link">
                        <span>Art & Culture</span>
                    </router-link>
                </template>
                    <a-menu-item key="symbols">
                        <span class="nav-sub">Symbols</span>
                    </a-menu-item>
                    <a-menu-item key="artist-of-day">
                        <router-link to="/artistofday" class="nav-sub">
                            <span>Artist of the Day</span>
                        </router-link>
                    </a-menu-item>
                    <a-menu-item key="art-facts">
                        <span class="nav-sub">Aboriginal Art Facts</span>
                    </a-menu-item>
            </a-sub-menu>
            <a-sub-menu>
                <template #title>
                    <router-link to="/exhibition" class="nav-link">
                        <span>Exhibitions</span>
                    </router-link>
                </template>
                <a-menu-item key="exhibition-currnt">
                    <router-link to="/exhibition" class="nav-sub">
                        <span>Current Exhibitions</span>
                    </router-link>
                </a-menu-item>
                <a-menu-item key="exhibition-past">
                    <span class="nav-sub">Past Exhibitions</span>
                </a-menu-item>
            </a-sub-menu>
            <a-sub-menu v-if="isAdmin()">
                <template #title>
                    <a href="https://localhost:7194/swagger/index.html" class="nav-link">Swagger</a>
                </template>
            </a-sub-menu>
        </a-menu>
    </div>
</template>

<script>
import { mapState } from 'vuex';
import { ref } from 'vue';

export default {
    name: "DropdownMenuAntD",
    methods: {
        isAdmin() {
            if (this.account.user) {
                return this.account.user.role == "Admin";
            }
        }
    },
    setup() {
        const current = ref(['mail']);
        return {
            current,
        };
    },
    computed: {
        ...mapState({
            account: state => state.account
        })
    }
}
</script>

<style scoped>
    .nav-sub {
        font-size: 16px;
        padding-left: 0;
    }
    .nav-sub:hover {
        color:var(--color--turquoise-light-hover);
        transition: .1s;
        transition-delay: 0;
        border-bottom: 2px solid;
    }

</style>
