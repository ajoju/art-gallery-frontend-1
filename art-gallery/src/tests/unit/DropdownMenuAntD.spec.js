import { mount } from "@vue/test-utils";
import DropdownMenuAntD from "@/components/DropdownMenuAntD.vue";

describe("DropdownMenuAntD", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(DropdownMenuAntD, {
      global: {
        mocks: {
          $store: {
            state: {
              account: {
                user: {
                  role: "Admin",
                },
              },
            },
          },
        },
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the Artworks sub-menu", () => {
    const artworksSubMenu = wrapper.find("a-sub-menu");
    expect(artworksSubMenu.exists()).toBe(true);
  });

  it("renders the Art & Culture sub-menu", () => {
    const artCultureSubMenu = wrapper.findAll("a-sub-menu")[1];
    expect(artCultureSubMenu.exists()).toBe(true);
  });

  it("renders the Exhibitions sub-menu", () => {
    const exhibitionsSubMenu = wrapper.findAll("a-sub-menu")[2];
    expect(exhibitionsSubMenu.exists()).toBe(true);
  });

  it("renders the Swagger sub-menu if user role is Admin", async () => {
    const swaggerSubMenu = wrapper.findAll("a-sub-menu")[3];
    expect(swaggerSubMenu.exists()).toBe(true);
  });

  it("does not render the Swagger sub-menu if user role is not Admin", async () => {
    wrapper = mount(DropdownMenuAntD, {
      global: {
        mocks: {
          $store: {
            state: {
              account: {
                user: {
                  role: "User",
                },
              },
            },
          },
        },
      },
    });

    const swaggerSubMenu = wrapper.find('a-sub-menu[title="Swagger"]');
    expect(swaggerSubMenu.exists()).toBe(false);
  });
});
