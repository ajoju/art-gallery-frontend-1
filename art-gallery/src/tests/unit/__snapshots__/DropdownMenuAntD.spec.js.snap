// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropdownMenuAntD > renders correctly 1`] = `
"<div data-v-55c2bd29=\\"\\" class=\\"nav-site\\">
  <a-menu data-v-55c2bd29=\\"\\" selectedkeys=\\"home\\" mode=\\"horizontal\\" class=\\"nav-menu\\" disabledoverflow=\\"{true}\\">
    <a-menu-item data-v-55c2bd29=\\"\\">
      <router-link data-v-55c2bd29=\\"\\" to=\\"/\\" class=\\"nav-link\\"><span data-v-55c2bd29=\\"\\">Home</span></router-link>
    </a-menu-item>
    <a-sub-menu data-v-55c2bd29=\\"\\">
      <a-menu-item data-v-55c2bd29=\\"\\" class=\\"menu-item\\">
        <router-link data-v-55c2bd29=\\"\\" to=\\"/artworkofday\\" class=\\"nav-sub\\"><span data-v-55c2bd29=\\"\\">Artwork of the Day</span></router-link>
      </a-menu-item>
      <a-menu-item data-v-55c2bd29=\\"\\">
        <router-link data-v-55c2bd29=\\"\\" to=\\"/artworks\\" class=\\"nav-sub\\"><span data-v-55c2bd29=\\"\\">List of Artworks</span></router-link>
      </a-menu-item>
    </a-sub-menu>
    <a-sub-menu data-v-55c2bd29=\\"\\">
      <a-menu-item data-v-55c2bd29=\\"\\"><span data-v-55c2bd29=\\"\\" class=\\"nav-sub\\">Symbols</span></a-menu-item>
      <a-menu-item data-v-55c2bd29=\\"\\">
        <router-link data-v-55c2bd29=\\"\\" to=\\"/artistofday\\" class=\\"nav-sub\\"><span data-v-55c2bd29=\\"\\">Artist of the Day</span></router-link>
      </a-menu-item>
      <a-menu-item data-v-55c2bd29=\\"\\"><span data-v-55c2bd29=\\"\\" class=\\"nav-sub\\">Aboriginal Art Facts</span></a-menu-item>
    </a-sub-menu>
    <a-sub-menu data-v-55c2bd29=\\"\\">
      <a-menu-item data-v-55c2bd29=\\"\\">
        <router-link data-v-55c2bd29=\\"\\" to=\\"/exhibition\\" class=\\"nav-sub\\"><span data-v-55c2bd29=\\"\\">Current Exhibitions</span></router-link>
      </a-menu-item>
      <a-menu-item data-v-55c2bd29=\\"\\"><span data-v-55c2bd29=\\"\\" class=\\"nav-sub\\">Past Exhibitions</span></a-menu-item>
    </a-sub-menu>
    <a-sub-menu data-v-55c2bd29=\\"\\"></a-sub-menu>
  </a-menu>
</div>"
`;
